src/main/app/controller/PostController.ts

```diff
@@ -146,11 +146,63 @@ export class PostController<T extends AnyObject> {
     //   const initData = { id: ' ', state: State.successAuthentication, serviceType: '', ...formData };
     //   req.session.userCase = initData;
     // }
+    const caseworkerUser = await getSystemUser();
+    req.locals.api = getCaseApi(caseworkerUser, req.locals.logger);
     req.session.errors = form.getErrors(formData);
+    const caseReference = formData.caseCode?.replace(/-/g, '');
+    try {
+      if (!req.session.errors.length) {
+        const caseData = await req.locals.api.getCaseById(caseReference as string);
+        let accessCodeMatched = false;
+        let accessCodeLinked = false;
+        if (caseData.respondentCaseInvites !== null) {
+          caseData.respondentCaseInvites?.forEach(obj => {
+            Object.entries(obj).forEach(([key, value]) => {
+              console.log(key);
+              Object.entries(value).forEach(([key1, value1]) => {
+                if (key1 === 'hasLinked' && value1 === 'Yes') {
+                  accessCodeLinked = true;
+                } else {
+                  accessCodeLinked = false;
+                }
+                if (key1 === 'accessCode' && value1 === formData.accessCode) {
+                  accessCodeMatched = true;
+                }
+              });
+            });
+          });
+        }
+        if (caseData.applicantCaseInvites !== null) {
+          caseData.applicantCaseInvites?.forEach(obj => {
+            Object.entries(obj).forEach(([key, value]) => {
+              console.log(key);
+              Object.entries(value).forEach(([key1, value1]) => {
+                if (key1 === 'hasLinked' && value1 === 'Yes') {
+                  accessCodeLinked = true;
+                } else {
+                  accessCodeLinked = false;
+                }
+                if (key1 === 'accessCode' && value1 === formData.accessCode) {
+                  accessCodeMatched = true;
+                }
+              });
+            });
+          });
+        }
+        if (!accessCodeMatched) {
+          req.session.errors.push({ errorType: 'invalidAccessCode', propertyName: 'accessCode' });
+        }
+        if (accessCodeLinked) {
+          req.session.errors.push({ errorType: 'accesscodeAlreadyLinked', propertyName: 'accessCode' });
+        }
+      }
+    } catch (err) {
```

Comment:
Is this going to be always InvalidReference error? can we not use actual error which we are catching in 'err'